js/ui/hierarchical_collection/ui.data_converter.js

```diff
@@ -169,7 +169,7 @@ var DataConverter = Class.inherit({
                 var currentElementKey = element.internalFields && element.internalFields.key || that._dataAccessors.getters.key(element),
                     items = that._dataAccessors.getters.items(element);
 
-                if(currentElementKey === key) {
+                if(currentElementKey.toString() === key.toString()) {
```

Comment:
We have the `toComparable` method, located in `core/utils/data`. Are any disadvantages to use it?

Reply:
toComparable("1") !== toComparable(1) but keys in the dxTreeView should be converted to one type